src/library/overviewcache.cpp

```diff
@@ -0,0 +1,217 @@
+#include "library/overviewcache.h"
+
+#include <QFutureWatcher>
+#include <QPixmapCache>
+#include <QSqlDatabase>
+#include <QtConcurrentRun>
+#include <QtSql>
+
+#include "library/dao/analysisdao.h"
+#include "moc_overviewcache.cpp"
+#include "util/db/dbconnectionpooled.h"
+#include "util/db/dbconnectionpooler.h"
+#include "util/logger.h"
+#include "waveform/renderers/waveformoverviewrenderer.h"
+#include "waveform/renderers/waveformsignalcolors.h"
+#include "waveform/waveformfactory.h"
+
+namespace {
+
+mixxx::Logger kLogger("OverviewCache");
+
+QString pixmapCacheKey(TrackId trackId, QSize size, mixxx::OverviewType type) {
+    return QString("Overview_%1_%2_%3_%4")
+            .arg(static_cast<int>(type))
+            .arg(trackId.toString())
+            .arg(size.width())
+            .arg(size.height());
+}
+
+// The transformation mode when scaling images
+const Qt::TransformationMode kTransformationMode = Qt::SmoothTransformation;
+
+inline QImage resizeImageSize(const QImage& image, QSize size) {
+    return image.scaled(size, Qt::IgnoreAspectRatio, kTransformationMode);
+}
+} // anonymous namespace
+
+OverviewCache::OverviewCache(UserSettingsPointer pConfig,
+        mixxx::DbConnectionPoolPtr pDbConnectionPool)
+        : m_pConfig(pConfig),
+          m_pDbConnectionPool(std::move(pDbConnectionPool)) {
+}
+
+void OverviewCache::onNormalizeOrVisualGainChanged() {
+    // TODO Lock to prevent interferences when this is called repeatedly when
+    // Normalize or VisualGainAll value are changed in quick succession?
+    // TODO use QMultihash::key_value_iterator to collect keys and values?
```

Comment:
do you want to do this checking / cleanup in this PR?

Reply:
I'm not sure how to proceed and if that's the best place to react on normalize/visual gain changes.
A locker here would basically ignore the last change notification.
Maybe we should listen to changes in OverviewDelegate and trigger a tracks view update. OverviewCache then reads the normalize/visual gain values, compares with the stored values and, if necessary, clears cached images and re-requests them one by one.

Reply:
I reworked this slot, please check if that makes sense.

Reply:
I regret that I have been promoted past the point of expertise in a lot of areas of the Mixxx codebase 😅 . This is a credit to y'all and the work that you've done to make the codebase better, but it means I'm not very good at reviewing the details any more.  

But, traditionally, if a signal can be pinged many times, the signals themselves should be stuffed into a queue of some sort that is processed on a ticker (could be 50ms, this is low priority) and can be deduped.

so:
* onNormalizeOrVisualGainChanged() -> spam calls into queue (necessary to lock queue, or use threadsafe queue)
* `while true; lock queue; dedupe queue without popping; pop 1 item and process it; unlock queue`

You have to protect against queue overload (first-run with 10000 tracks being processed), so I would say only process things that are visible.

Reply:
Haha...

Thanks for the advice!
This was rather about identical "changed" signals, either from the gain spinbox or the normalize checkbox in the preferences.
So, yeah, a queue/wait/discard mechanism like here https://github.com/mixxxdj/mixxx/pull/11708/commits/abfcc99699a61ae2076865b7564663ce06926c8c would suffice probably.